src/widgets/buttons.ts

```diff
@@ -478,6 +488,42 @@ export function deleteButtonWithCheck (
   return deleteButtonElt
 }
 
+function getButtonStyle (options: ButtonWidgetOptions = { buttonType: ButtonType.Primary, filled: true }) {
+  // We need to accomadate for legacy code, which is why we have to allow buttonType and filled to be optional
+  let backgroundColor: (ButtonType | string | undefined) = options.buttonType
+  let fontColor: (ButtonType | string | undefined) = '#ffffff'
+  let borderColor: (ButtonType | string | undefined) = options.buttonType
+  let hoverBackgroundColor: (ButtonType | string | undefined) = `lighten(${options.buttonType}, 5%)`
+  let hoverFontColor: (ButtonType | string | undefined) = fontColor
+  if (!options.filled) {
```

Comment:
I went back and forth with this.  At one point I had let setting defaults and then have if (filled), else but decided it was actually adding much more code than necessary and maybe the better approach would be to just default to what the values would be if the option filled were chosen.

Reply:
yeah, this is fine. my only remark is it's best practice to name your options so that the default is negative.

So here you have filled vs. borderOnly.

If you want filled to be the default, you would call your option variable borderOnly (default: false).
If you want borderOnly to be the default, you would call your option variable filled (default: false).

That makes the code easier to reason about without having to deal with double negations in your head.

Reply:
that sounds good.